tests/gql_behave/tests/memgraph_V1/features/memgraph_allshortest.feature

```diff
@@ -203,3 +203,79 @@ Feature: All Shortest Path
+          """
+      Then the result should be:
+          | path                                                                                        | total_weight   |
+          | <(:label1 {id: 1})-[:type1 {id: 1}]->(:label2 {id: 2})-[:type1 {id: 2}]->(:label3 {id: 3})> | 3              |
+
+    Scenario: Test match AllShortest with accumulated path filtered by edge type1 and accumulated weight based on edge
+      Given graph "graph_edges"
+      When executing query:
+          """
+          MATCH path=(:label1)-[*ALLSHORTEST (r, n | r.id) total_weight (e, n, p, w | NOT(type(e)='type1' AND type(last(relationships(p))) = 'type1') AND w > 0)]->(:label3) RETURN path, total_weight;
+          """
+      Then the result should be:
+          | path                                                     | total_weight   |
+          | <(:label1 {id: 1})-[:type2 {id: 10}]->(:label3 {id: 3})> | 10             |
+
+    Scenario: Test match AllShortest with accumulated path filtered by edge type1 and accumulated weight based on edge too restricted
+      Given graph "graph_edges"
+      When executing query:
+          """
+          MATCH path=(:label1)-[*ALLSHORTEST (r, n | r.id) total_weight (e, n, p, w | NOT(type(e)='type1' AND type(last(relationships(p))) = 'type1') AND w < 10)]->(:label3) RETURN path, total_weight;
+          """
+      Then the result should be empty
+
+    Scenario: Test match AllShortest with accumulated path filtered by edge type1 and accumulated weight based on vertex is int
+      Given graph "graph_edges"
+      When executing query:
+          """
+          MATCH path=(:label1)-[*ALLSHORTEST (r, n | n.id) total_weight (e, n, p, w | NOT(type(e)='type1' AND type(last(relationships(p))) = 'type1') AND w > 0)]->(:label3) RETURN path, total_weight;
+          """
+      Then the result should be:
+          | path                                                     | total_weight   |
+          | <(:label1 {id: 1})-[:type2 {id: 10}]->(:label3 {id: 3})> | 4              |
+
+    Scenario: Test match AllShortest with accumulated path filtered by order of ids and accumulated weight based on both vertex and edge is duration
+      Given an empty graph
+      And having executed:
+          """
+          CREATE (:station {name: "A", arrival: localTime("08:00"), departure: localTime("08:15")})-[:ride {id: 1, duration: duration("PT1H5M")}]->(:station {name: "B", arrival: localtime("09:20"), departure: localTime("09:30")})-[:ride {id: 2, duration: duration("PT30M")}]->(:station {name: "C", arrival: localTime("10:00"), departure: localTime("10:20")});
+          """
+      When executing query:
+          """
+          MATCH path=(:station {name:"A"})-[*ALLSHORTEST (r, v | v.departure - v.arrival + r.duration) total_weight (r,n,p,w | (nodes(p)[-1]).name > (nodes(p)[-2]).name AND not(w is null))]->(:station {name:"C"}) RETURN path, total_weight;
```

Comment:
oh so actually the path lambda first time executes when we have 2 nodes in it, right?

Reply:
First call of path filter lambda is done on expansion to check if we can expand from one node to another. So yes, we have 2 nodes in path in this case.